### PR TITLE
Add syntax page to Cypher manual

### LINKED ISSUE #155: Document keywords and whether or not they're reserved
As of [3626](https://github.com/kuzudb/kuzu/pull/3626), we've decided that the keyword list should be documented, as well as whether or not each keyword is reserved. Additionally, we should document whether or not the status of the keyword complies with whether or not it's reserved in Postgres. In the following table, NA means the keyword doesn't exist in PG.
|Keyword|Status|Complies with PG?|
|-|-|-|
|ADD|Not Reserved|Complies|
|ALL|Reserved|Complies|
|ALTER|Not Reserved|Complies|
|AND|Reserved|Complies|
|ASCENDING|Reserved|NA|
|ASC|Reserved|Does Not Comply|
|AS|Not Reserved|Complies|
|ATTACH|Not Reserved|Complies|
|BEGIN|Not Reserved|Complies|
|BY|Not Reserved|Complies|
|CALL|Not Reserved|Complies|
|CASE|Reserved|Complies|
|CAST|Reserved|Complies|
|COLUMN|Reserved|Complies|
|COMMENT|Not Reserved|Complies|
|COMMIT\_SKIP\_CHECKPOINT|Reserved|NA|
|COMMIT|Not Reserved|Complies|
|CONTAINS|Not Reserved|NA|
|COPY|Not Reserved|Complies|
|COUNT|Not Reserved|NA|
|CREATE|Reserved|Complies|
|CYCLE|Not Reserved|Complies|
|DATABASE|Not Reserved|Complies|
|DBTYPE|Reserved|NA|
|DECIMAL|Not Reserved|Complies|
|DEFAULT|Reserved|Complies|
|DELETE|Not Reserved|Complies|
|DESCENDING|Reserved|NA|
|DESC|Reserved|Complies|
|DETACH|Not Reserved|Complies|
|DISTINCT|Reserved|Complies|
|DROP|Not Reserved|Complies|
|ELSE|Reserved|Complies|
|END|Reserved|Complies|
|ENDS|Reserved|NA|
|EXISTS|Reserved|NA|
|EXPLAIN|Not Reserved|Complies|
|EXPORT|Not Reserved|NA|
|EXTENSION|Not Reserved|Complies|
|FALSE|Reserved|Complies|
|FROM|Reserved|Complies|
|GLOB|Reserved|NA|
|GRAPH|Not Reserved|NA|
|GROUP|Reserved|Complies|
|HEADERS|Reserved|NA|
|IF|Not Reserved|Complies|
|IMPORT|Not Reserved|Complies|
|INCREMENT|Not Reserved|Complies|
|IN|Reserved|Complies|
|INSTALL|Reserved|NA|
|IS|Reserved|NA|
|KEY|Not Reserved|Complies|
|LIMIT|Reserved|Complies|
|LOAD|Not Reserved|Complies|
|MACRO|Reserved|NA|
|MATCH|Not Reserved|Complies|
|MAXVALUE|Not Reserved|Complies|
|MERGE|Not Reserved|Complies|
|MINVALUE|Not Reserved|Complies|
|NODE|Not Reserved|NA|
|NOT|Reserved|Complies|
|NO|Not Reserved|Complies|
|NULL|Reserved|NA|
|ONLY|Reserved|Complies|
|ON|Reserved|Complies|
|OPTIONAL|Reserved|NA|
|ORDER|Reserved|Complies|
|OR|Reserved|Complies|
|PRIMARY|Reserved|Complies|
|PROFILE|Reserved|NA|
|PROJECT|Not Reserved|Complies|
|RDFGRAPH|Reserved|NA|
|READ|Not Reserved|Complies|
|REL|Not Reserved|NA|
|RENAME|Not Reserved|Complies|
|RETURN|Not Reserved|Complies|
|ROLLBACK\_SKIP\_CHECKPOINT|Reserved|NA|
|ROLLBACK|Not Reserved|Complies|
|SEQUENCE|Not Reserved|Complies|
|SET|Not Reserved|Complies|
|SHORTEST|Reserved|NA|
|STARTS|Reserved|AN|
|START|Not Reserved|Complies|
|TABLE|Reserved|Complies|
|THEN|Reserved|Complies|
|TO|Reserved|Complies|
|TRANSACTION|Not Reserved|Complies|
|TRUE|Reserved|Complies|
|TYPE|Not Reserved|Complies|
|UNION|Reserved|Complies|
|UNWIND|Reserved|NA|
|USE|Not Reserved|NA|
|WHEN|Reserved|Complies|
|WHERE|Reserved|Complies|
|WITH|Reserved|Complies|
|WRITE|Not Reserved|Complies|
|XOR|Reserved|NA|

plaintext for the markdown table can be accessed by editing this issue.

### LINKED ISSUE #196: Describe string escape rules in the docs
We should add string escape rules similar to https://www.postgresql.org/docs/current/sql-syntax-lexical.html#SQL-SYNTAX-STRINGS and https://dev.mysql.com/doc/refman/8.4/en/string-literals.html. 

If our escape rules are the same as an existing database, we should add a link to their docs and state that our escape rules are the same.

Together with https://github.com/kuzudb/kuzu/issues/3975, this will be helpful when the user is working with string literals (e.g. writing a program that generates queries from raw strings).

### LINKED ISSUE #238: Document cypher comment styles
We support adding comments in cypher queries, but not documented yet.

### PR DESCRIPTION
**Pull request by prrao87**
This page addresses some open issues related to the Cypher manual's docs. I've created a new page `syntax` that documents these. It describes the key aspects of Cypher syntax for new users.

See the below issues it closes.

- [x] Reserved keywords: Closes #155 
- [x] String escape rules: Closes #196 
- [x] Comments in Cypher: Closes #238

### CODE REVIEW

**Comment by prrao87** [comment]
@andyfengHKU I documented the reserved keywords and categorized them to the best of my ability. I have no strong opinions on the categories of each reserved keyword, so if you think some of them are wrong, please move around as needed. 🙏🏽

File: src/content/docs/cypher/syntax.md
Comment (andyfengHKU) [inline src/content/docs/cypher/syntax.md:10]: Parsing sounds a bit too abstract for this section.

I would split them into 
```
Encoding (discuss ascii, unicode, ...)

Escaping (discuss special character, using space as a table name, etc)

Multiline statement (discuss line breaking and ; termination)
```
But this is up to you.

File: src/content/docs/cypher/syntax.md
Comment (andyfengHKU) [inline src/content/docs/cypher/syntax.md:42]: Is this from Neo4j, I have a feeling that user probably doesn't need to understand the terminology of Clause.

If we decide to keep this section, is `subclause` a defined notion by Neo4j? I don't recall seeing this in the openCypher manual.

File: src/content/docs/cypher/syntax.md
Comment (andyfengHKU) [inline src/content/docs/cypher/syntax.md:68]: I recall we had an issue here on the CLI end. Can u try and open an issue if necessary? If we do have a bug, let's not document this example.

File: src/content/docs/cypher/syntax.md
Comment (andyfengHKU) [inline src/content/docs/cypher/syntax.md:116]: First I think we should call this section keyword. And inside we have reserved and non-reserved keyword.

We will highlight all keywords. Non-reserved ones can still used as variables.

keyword list is in `src/antlr/keywords.txt`
non-reserved ones is in `src/antlr/Cypher.g4` and rule `kU_NonReservedKeywords`

I would just follow [PG](https://www.postgresql.org/docs/current/sql-keywords-appendix.html) and organize them in alphabet order

**Review by andyfengHKU — APPROVED** [review]
Overall LGTM. I left a few suggestions but the final decision is up to u.

File: src/content/docs/cypher/syntax.md
Comment (prrao87) [reply src/content/docs/cypher/syntax.md:42]: I think clause is reasonably well understood from its role in natural language sentences, too. It's not just for query languages, so I think we can leave it there. I broke it down by section because it can be beneficial for users to know what category belongs to.

And subclause is in openCypher9, see the "Reserved keywords" section. It's also adopted by Neo4j, but I looked at the openCypher9 guide.
https://s3.amazonaws.com/artifacts.opencypher.org/openCypher9.pdf

File: src/content/docs/cypher/syntax.md
Comment (prrao87) [reply src/content/docs/cypher/syntax.md:10]: I used the same subheader that Neo4j did in the start of their Cypher manual, under syntax. But ours is less verbose and all in one page (Neo4j breaks theirs down into multiple pages).

File: src/content/docs/cypher/syntax.md
Comment (prrao87) [reply src/content/docs/cypher/syntax.md:116]: Is it just as important to list non-reserved keywords? if it's not reserved, why should the user be told that it can be used as variables? This is redundant and expects the user to do more work, imo. Also, I agree with Semih in that PG shouldn't be brought in here unless absolutely necessary (it creates more confusion because this is for a Cypher manual).

File: src/content/docs/cypher/syntax.md
Comment (prrao87) [reply src/content/docs/cypher/syntax.md:68]: I'm waiting for the bug to be resolved before this is merged :). It currently works in Python API and Kùzu Explorer, I just tested there. But for CLI, it isn't working yet. Will check if there's an issue, and if not, will post one.

File: src/content/docs/cypher/syntax.md
Comment (prrao87) [reply src/content/docs/cypher/syntax.md:116]: Again, this section is designed to be more or less identical to the openCypher9 PDF:
https://s3.amazonaws.com/artifacts.opencypher.org/openCypher9.pdf

Comment (prrao87) [reply src/content/docs/cypher/syntax.md:116]: > We will highlight all keywords.
This point, I agree, makes sense. I'll think a bit more about how/where to list all available keywords. But I think the reserved keywords subheader should be present as this is much more important for the reader.

File: src/content/docs/cypher/syntax.md
Comment (prrao87) [reply src/content/docs/cypher/syntax.md:10]: Ok, fixed this by adding more subheadings to the "Parsing section".

**Pull request merged**
